src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -36,12 +93,68 @@ public InnReachResponseDTO createInnReachTransactionItemHold(String trackingId,
     return response;
   }
 
-  private InnReachTransaction createTransactionWithItemHold(String trackingId, String centralCode) {
-    InnReachTransaction transaction = new InnReachTransaction();
-    transaction.setTrackingId(trackingId);
-    transaction.setCentralServerCode(centralCode);
-    transaction.setType(TransactionType.ITEM);
-    transaction.setState(InnReachTransaction.TransactionState.ITEM_HOLD);
-    return transaction;
+  @Override
+  public void createItemRequest(String trackingId) {
+    var transaction = repository.fetchOneByTrackingId(trackingId).orElseThrow(() ->
+      new EntityNotFoundException("Transaction with trackingId = " + trackingId + " not found.")
+    );
+    var itemId = transaction.getHold().getItemId();
+    String errorReason;
+
+    var item = inventoryClient.getItemById(itemId);
+    var requests = requestsClient.findRequests(itemId);
+
+    if (itemIsRequestable(item, requests)) {
+      try {
+        var comment = "INN-Reach request: Patron Agency - " + transaction.getHold().getPatronAgencyCode() +
```

Comment:
Could you simplify this block of code (split into one or several private methods), it's not easy to read.